datafusion/expr/src/expr.rs

```diff
@@ -419,6 +419,11 @@ impl ScalarFunction {
             args,
         }
     }
+
+    /// Create a new ScalarFunction expression with a user-defined function (UDF)
+    pub fn new_func_def(func_def: ScalarFunctionDefinition, args: Vec<Expr>) -> Self {
```

Comment:
Is it better to inline the arguments directly?

Reply:
Could do that, sure.